sdk/attestation/azure-security-attestation/inc/azure/attestation/attestation_client.hpp

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <azure/core/internal/json/json.hpp>
```

Comment:
That's going to be ... interesting for internal headers.

Reply:
All parsing is done in headers, not cpp files. That allows the compiler to do a better job of inlining (LTCG and comdat folding helps but putting the code in headers is generally better, IMHO).

Reply:
We've had issues in the past by having implementation in headers.

Something related to dynamic linking.  There were some limits about what to do implement and what to avoid implementing on headers.

There was also something related to making compilation time longer.

Anyway, you could create a private header (within the src/ folder) and add inline implementation to it.  Then consume the file from the client.cpp implementation.  But avoid bringing the json header to public client, since the json header is huge (it's all the lib in a header).

Reply:
Raw functions in headers breaks when you have headers in multiple files. Putting the functions in a class resolves that issue. 

And my parsing functions are in a private header - they're COMPLETELY not a part of the public contract.